### DISCUSSION TITLE
Removing tokens from a doc object

### DISCUSSION THREAD

**Discussion by ghost** [explosion/spaCy#10765] [Help: Coding & Implementations] [answered]
Hello,

I am working on a text data that contain bullet points. For example, "1. The Day is tuesday " . And I want to remove the number and the punctuation next to it so that the tokens begin with "The"
Can you please suggest what I can do ?

**Comment by polm** [comment, marked as answer]
You can't modify the text of existing Docs, that's a design decision. 

That leaves you basically two options. 

1. Preprocess the text before it gets to spaCy
2. Use spaCy to create a Doc, then filter the tokens to create a new Doc

For option 1 you can use a regex or something.

For option 2 you could use the Matcher or other techniques to identify your bullet points. You could then use [`Span.as_doc`](https://spacy.io/api/span#as_doc) or [the Doc constructor](https://spacy.io/api/doc#init) to create new Docs.

Out of curiosity, have the bullet points actually caused problems for you so far? I would be surprised if they had a large effect on models, and would expect it to be fine to leave them in and filter them out afterwards.

**Reply by ghost** [reply]
Thanks!
Actually, I was implementing the tf-idf vectorization using the doc objects from token.text. I thought it was better to remove the bullets !
